File: vortexdbg-android/src/main/java/com/vortexdbg/linux/thread/PThreadInternal.java
package com.vortexdbg.linux.thread;

import com.vortexdbg.Emulator;
import com.vortexdbg.pointer.VortexdbgStructure;
import com.sun.jna.Pointer;

public abstract class PThreadInternal extends VortexdbgStructure {

    public static PThreadInternal create(Emulator<?> emulator, Pointer pointer) {
        return emulator.is64Bit() ? new PThreadInternal64(pointer) : new PThreadInternal32(pointer);
    }

    public int tid;

    public PThreadInternal(Pointer p) {
        super(p);
    }

}

File: vortexdbg-android/src/main/kotlin/com/vortexdbg/linux/thread/PThreadInternal.kt
package com.vortexdbg.linux.thread

import com.vortexdbg.Emulator
import com.vortexdbg.pointer.VortexdbgStructure
import com.sun.jna.Pointer

abstract class PThreadInternal(p: Pointer?) : VortexdbgStructure(p) {

    @JvmField
    var tid: Int = 0

    companion object {
        @JvmStatic
        fun create(emulator: Emulator<*>, pointer: Pointer): PThreadInternal {
            return if (emulator.is64Bit()) PThreadInternal64(pointer) else PThreadInternal32(pointer)
        }
    }

}
